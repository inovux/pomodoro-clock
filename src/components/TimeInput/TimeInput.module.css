.container {
  display: flex;
}

.time {
  display: inline-block;
  font-family: 'Open Sans', arial, sans-serif;
  font-size: 2.2rem;
  font-weight: 600;
  line-height: 1.4;
  text-align: center;
  color: #626262;
  width: 6rem;
  cursor: text;
  padding: 1px 1px 0px 1px;
  border-bottom: 2px solid transparent;
}

.timeInput {
  outline: none;
  border: none;
  background-color: transparent;
  border-bottom: 2px solid #626262;
}
