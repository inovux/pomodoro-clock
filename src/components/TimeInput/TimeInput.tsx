import React, { useState } from 'react';
import classNames from 'classnames';

import styles from './TimeInput.module.css';
import Button from '../Button';

type Props = {
  onChange: (value: number) => void;
  value: number;
};

const TimeInput = ({ onChange, value }: Props) => {
  const [isEditable, setIsEditable] = useState(false);

  const onChangeInput = (e: React.FormEvent<HTMLInputElement>) => {
    const re = /^[0-9\b]+$/;

    if (e.currentTarget.value === '' || re.test(e.currentTarget.value)) {
      onChange(+e.currentTarget.value);
    }
  };

  const onIncrementTime = () => {
    onChange(value + 1);
  };

  const onDecrementTime = () => {
    onChange(value - 1);
  };

  return (
    <div className={styles.container}>
      <Button onClick={() => onDecrementTime()}>{'<<'}</Button>
      {isEditable ? (
        <input
          className={classNames(styles.timeInput, styles.time)}
          type="text"
          onChange={onChangeInput}
          value={value}
          autoFocus
          onBlur={() => setIsEditable(false)}
        />
      ) : (
        <span className={styles.time} onClick={() => setIsEditable(true)}>
          {value}
        </span>
      )}
      <Button onClick={() => onIncrementTime()}>{'>>'}</Button>
    </div>
  );
};

export default TimeInput;
