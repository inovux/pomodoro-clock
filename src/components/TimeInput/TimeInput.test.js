import React from 'react';
import { render, cleanup } from '@testing-library/react';

import TimeInput from './TimeInput';

afterEach(cleanup);

describe('TimeInput', () => {
  it('renders', () => {
    const { asFragment } = render(<TimeInput />);

    expect(asFragment()).toMatchSnapshot();
  });
});
