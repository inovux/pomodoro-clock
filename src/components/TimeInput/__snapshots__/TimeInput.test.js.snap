// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeInput renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button
      class="container"
    >
      &lt;&lt;
    </button>
    <span
      class="time"
    >
      0
    </span>
    <button
      class="container"
    >
      &gt;&gt;
    </button>
  </div>
</DocumentFragment>
`;
